### DISCUSSION TITLE
Use JupyterLab Web IDE with Coder

### DISCUSSION THREAD

**Discussion by framctr** [coder/coder#13880] [General] [answered]
Hi,
I tried to launch Jupyter with Coder following the [official documentation](https://coder.com/docs/@v0.8.1/ides/configuring-web-ides), but it didn't work and I got 404 errors.
After trying different configurations, I found out a solution and I decided to post it here for other users who may find the same issue.
First the documentation seems a bit old and some parameters are deprecated. I found out that when launching Jupyter the `base_url` and the `coder_app` URL must be the same, but in addition the `coder_app` must specify the `slug = "jupyter"`.

```hcl
data "coder_workspace" "me" {}
data "coder_workspace_owner" "me" {} # use this since `data.coder_workspace.me.owner` was deprecated

resource "coder_agent" "main" {
  os   = "linux"
  arch = "amd64"
  dir  = "/home/coder"

  startup_script = <<-EOT
    set -e

    pip3 install jupyterlab

    # install and start JupyterLab
    ~/.local/bin/jupyter lab --ServerApp.base_url=/@${data.coder_workspace_owner.me.name}/${data.coder_workspace.me.name}/apps/jupyter --ServerApp.token='' --ip='*'
  EOT
}

## As the Coder documentation says:
## The name of the app must always be equal to the "/apps/<name>"
## string in the base_url. This caveat is unique to Jupyter.

resource "coder_app" "jupyter" {
  agent_id     = coder_agent.main.id
  display_name = "JupyterLab"
  icon         = "/icon/jupyter.svg"
  url          = "http://localhost:8888/@${data.coder_workspace_owner.me.name}/${data.coder_workspace.me.name}/apps/jupyter"
  slug         = "jupyter" # keep the same as the end of URL
  subdomain    = false
  share        = "owner"
}
```

Happy Codering!

**Comment by matifali** [comment, marked as answer]
Looks like you are looking at older documentations. This no more missing in latest docs.
https://coder.com/docs/ides/web-ides#jupyterlab

We also publish a teraform module to add jupyterlab to any template.

https://registry.coder.com/modules/jupyterlab
